### PR TITLE
Implement Listing Pattern in base package

### PR DESCRIPTION
**Pull request by stefwalter**
This is based on the images that the Patternfly team has sent us.

We aren't yet using a close button, but the other styling changes have been applied.

### CODE REVIEW

**Comment by stefwalter** [comment]
Time line is based on: https://raw.githubusercontent.com/cockpit-project/cockpit-design/master/software-updates/software-updates-ostree-alt.png

**Comment by stefwalter** [comment]
Sent to us by Patternfly folks: 

cockpit-list-expand_2015-09-24

**Comment by stefwalter** [comment]
Check out the pattern in the 'Design Patterns' section of Cockpit

**Comment by stefwalter** [comment]
Updated cockpit-design for new styling in the patterns folder:

image

**Comment by stefwalter** [comment]
@serenamarie125 @kybaker FYI ... this is implementing most of the listing pattern changes. In HTML, CSS jQUery and in our angular code as well.

**Comment by petervo** [comment]
The "Expandable Listing: Panel header is row" version might little a little more padding between the blue box and the tab text when hovering over an expanded row. 

Other than that seems good to me

**Comment by stefwalter** [comment]
> The "Expandable Listing: Panel header is row" version might little a little more padding between the blue box and the tab text when hovering over an expanded row.

I'd like to fix that in javascript ... so the whole header hovers. Not possible in CSS, but good catch will fix in the JS.

**Comment by stefwalter** [comment]
For the record, the logic behind the different colored headers:

 1. gray header is used with non-collapsable panels
 2. white body visually blends too easily into following collapsed rows
 3. for exapandable panels, white header matches the row you just clicked on

**Comment by stefwalter** [comment]
I had fixed the build issue, but forgot to push again. Fingers crossed.

**Comment by stefwalter** [comment]
Fixed up check-openshift and check-kubernetes tests.

**Comment by stefwalter** [comment]
r7 test failure is #2906

**Comment by stefwalter** [comment]
Rebased on #2906

File: pkg/base1/cockpit.css [hunk 1]
```diff
@@ -548,3 +548,396 @@ a {
...
+    padding: 5px 10px;
+    text-align: right;
+    float: none;
+}
+
+tr.listing-item td:first-child {
+    padding-left: 20px;
+}
+
+/* The last column of a listing is always rigth aligned */
```
Comment (dperpeet) [inline pkg/base1/cockpit.css]: typo: right

File: pkg/base1/cockpit.css [hunk 2]
```diff
@@ -548,3 +548,396 @@ a {
...
+    font-size: 22px;
+    padding: 10px 0px 10px;
+    font-weight: 300;
+}
+
+table.listing > thead:first-child h3 {
+    margin-top: 0px;
+}
+
+
```
Comment (dperpeet) [inline pkg/base1/cockpit.css]: extra whitespace

File: pkg/kubernetes/app.js [hunk 3]
```diff
@@ -162,15 +162,17 @@ define([
                             }
                         };
 
-                        scope.select = function select(id, value) {
+                        scope.select = function select(id, ev) {
+                            var value;
+                            if (ev && $(ev.target).parents().andSelf().filter(".btn, li").length > 0)
```
Comment (dperpeet) [inline pkg/kubernetes/app.js]: [andSelf() is deprecated](https://api.jquery.com/andself/)
such a long chain also might deserve a comment

File: pkg/kubernetes/views/container-panel.html [hunk 4]
```diff
@@ -1,33 +1,28 @@
...
-                </kube-shell>
-            </div>
-        </tab>
-    </tabset>
+    <h3><i class="fa fa-cube"></i>{{container.spec.name}}</h3>
+    <div class="listing-status">
+        <span ng-repeat="(state, value) in container.status.state">{{state}}</span>
+    </div>
+    <ul class="nav nav-tabs nav-tabs-pf">
+        <li ng-class="{active: tab('main')}"><a ng-click="tab('main', $event)">Container</a></li>
```
Comment (dperpeet) [inline pkg/kubernetes/views/container-panel.html]: translatable

File: pkg/kubernetes/views/container-panel.html [hunk 5]
```diff
@@ -1,33 +1,28 @@
...
+        <li ng-class="{active: tab('logs')}">
+            <a ng-click="tab('logs', $event); connect('console')">Logs</a></li>
```
Comment (dperpeet) [inline pkg/kubernetes/views/container-panel.html]: translatable

File: pkg/kubernetes/views/container-panel.html [hunk 6]
```diff
@@ -1,33 +1,28 @@
...
+        <li ng-class="{active: tab('shell')}" ng-if="container.status.state.running">
+            <a ng-click="tab('shell', $event); connect('shell')">Shell</a></li>
```
Comment (dperpeet) [inline pkg/kubernetes/views/container-panel.html]: translatable

File: pkg/kubernetes/views/containers-page.html [hunk 7]
```diff
@@ -1,10 +1,12 @@
...
-    <table class="listing containers-listing">
-        <thead>
-            <tr class="listing-title">
-                <td colspan="5" translatable="yes">Containers</td>
+<div>
+    <table listing-table class="listing containers-listing">
+        <thead ng-if="!quiet">
+            <tr>
+                <td colspan="4">
+                    <h3>Containers</h3>
```
Comment (dperpeet) [inline pkg/kubernetes/views/containers-page.html]: translatable

File: pkg/kubernetes/views/containers-page.html [hunk 8]
```diff
@@ -13,19 +15,23 @@
                 <th translatable="yes">Status</th>
             </tr>
         </thead>
-        <tbody ng-repeat="item in pods | orderObjectBy:['metedata.namespace', 'metadata.name']">
-            <tr ng-repeat-start="(id, container) in containers(item)"
-                ng-click="select(id)" ng-if="!quiet && !selected(id)">
+        <tbody data-ng-bullshit="" ng-if="0"
```
Comment (dperpeet) [inline pkg/kubernetes/views/containers-page.html]: do we need to use the word "bullshit" in our code?

File: pkg/kubernetes/views/containers-page.html [hunk 9]
```diff
@@ -13,19 +15,23 @@
...
                 <th>{{container.spec.name}}</th>
                 <td>{{item.metadata.name}}</td>
                 <td>{{item.metadata.namespace}}</td>
                 <td>{{item.spec.nodeName}}</td>
                 <td><span ng-repeat="(name, value) in container.status.state">{{name}}</span></td>
             </tr>
-            <tr ng-repeat-end class="listing-panel" ng-if="selected(id)">
-                <td cockpit-listing-panel kind="Container" colspan="5">
+            <tr class="listing-panel" ng-if="selected(id)">
+                <td listing-panel kind="Container" colspan="5">
```
Comment (dperpeet) [inline pkg/kubernetes/views/containers-page.html]: in the lines above, you close <td> on the same line

File: pkg/kubernetes/views/containers-page.html [hunk 10]
```diff
@@ -13,19 +15,23 @@
...
                 </td>
             </tr>
         </tbody>
+        <tbody data-ng-bullshit="" ng-if="0" ng-repeat-end>
```
Comment (dperpeet) [inline pkg/kubernetes/views/containers-page.html]: see earlier comment

File: pkg/kubernetes/views/deploymentconfig-panel.html [hunk 11]
```diff
@@ -1,66 +1,68 @@
...
-        <div class="listing-icons">
-            <i class="fa fa-gear"></i>
-        </div>
-        <div class="listing-actions">
-            <button title="Delete" class="btn btn-danger btn-delete pficon pficon-delete" disabled></button>
-        </div>
-        <h3 class="listing-title" ng-click="select(id)">{{ item.metadata.name }}</h3>
+<div class="listing-head" ng-click="select(id, $event)">
+    <div class="listing-actions">
+        <button title="Delete" class="btn btn-danger btn-delete pficon pficon-delete" disabled></button>
```
Comment (dperpeet) [inline pkg/kubernetes/views/deploymentconfig-panel.html]: translatable

File: pkg/kubernetes/views/deploymentconfig-panel.html [hunk 12]
```diff
@@ -1,66 +1,68 @@
...
-                            <span>
-                        </dd>
+    <h3><i class="fa fa-gear"></i>{{ item.metadata.name }}</h3>
+    <div class="listing-status">
+        {{item.status.conditions | nodeStatus}}
+    </div>
+    <ul class="nav nav-tabs nav-tabs-pf">
+        <li ng-class="{active: tab('main')}"><a ng-click="tab('main', $event)">Deployment Config</a></li>
+        <li ng-class="{active: tab('template')}" ng-if="item.spec.template">
+            <a ng-click="tab('template', $event)">Template</a></li>
```
Comment (dperpeet) [inline pkg/kubernetes/views/deploymentconfig-panel.html]: translatable

File: pkg/kubernetes/views/deploymentconfig-panel.html [hunk 13]
```diff
@@ -1,66 +1,68 @@
...
-                            <span ng-if="cause.imageTrigger.from">
-                                - {{ cause.imageTrigger.from.name }}
-                            <span>
-                        </dd>
+    <h3><i class="fa fa-gear"></i>{{ item.metadata.name }}</h3>
+    <div class="listing-status">
+        {{item.status.conditions | nodeStatus}}
+    </div>
+    <ul class="nav nav-tabs nav-tabs-pf">
+        <li ng-class="{active: tab('main')}"><a ng-click="tab('main', $event)">Deployment Config</a></li>
```
Comment (dperpeet) [inline pkg/kubernetes/views/deploymentconfig-panel.html]: translatable

File: pkg/kubernetes/views/details-page.html [hunk 14]
```diff
@@ -1,8 +1,10 @@
-<div cockpit-listing>
-    <table class="listing details-listing">
+<div>
+    <table listing-table class="listing details-listing">
         <thead ng-if="!quiet">
-            <tr class="listing-title">
-                <td colspan="4" translatable="yes">Nodes</td>
+            <tr>
+                <td colspan="4">
+                    <h3>Nodes</h3>
```
Comment (dperpeet) [inline pkg/kubernetes/views/details-page.html]: translatable

File: pkg/kubernetes/views/details-page.html [hunk 15]
```diff
@@ -53,14 +57,16 @@
...
                 </td>
             </tr>
         </tbody>
 
         <thead id="routes" ng-if="!quiet && client.flavor == 'openshift'">
-            <tr class="listing-title">
-                <td colspan="4" translatable="yes">Routes</td>
+            <tr>
+                <td colspan="4">
+                    <h3>Routes</h3>
```
Comment (dperpeet) [inline pkg/kubernetes/views/details-page.html]: translatable

File: pkg/kubernetes/views/details-page.html [hunk 16]
```diff
@@ -71,22 +77,25 @@
...
                 </td>
             </tr>
         </tbody>
 
         <thead id="deployment-configs" ng-if="!quiet && client.flavor == 'openshift'">
-            <tr class="listing-title">
-                <td colspan="4" translatable="yes">Deployment Configs</td>
+            <tr>
+                <td colspan="4">
+                    <h3>Deployment Configs</h3>
```
Comment (dperpeet) [inline pkg/kubernetes/views/details-page.html]: translatable

File: pkg/kubernetes/views/details-page.html [hunk 17]
```diff
@@ -106,16 +116,16 @@
                 <td ng-if="!item.status.latestVersion" translatable="yes">Not deployed</td>
             </tr>
             <tr class="listing-panel" ng-if="selected(id)">
-                <td cockpit-listing-panel kind="DeploymentConfig" colspan="4">
+                <td listing-panel kind="DeploymentConfig" colspan="4">
```
Comment (dperpeet) [inline pkg/kubernetes/views/details-page.html]: close tag in same line?

File: pkg/kubernetes/views/details-page.html [hunk 18]
```diff
@@ -106,16 +116,16 @@
...
         </tbody>
 
-        <thead>
-
         <thead ng-if="!quiet">
-            <tr class="listing-title">
-                <td colspan="4" translatable="yes">Replication Controllers</td>
+            <tr>
+                <td colspan="4">
+                    <h3>Replication Controllers</h3>
```
Comment (dperpeet) [inline pkg/kubernetes/views/details-page.html]: translatable

File: pkg/kubernetes/views/details-page.html [hunk 19]
```diff
@@ -136,14 +146,16 @@
...
                 </td>
             </tr>
         </tbody>
 
         <thead ng-if="!quiet">
-            <tr class="listing-title">
-                <td colspan="4" translatable="yes">Pods</td>
+            <tr>
+                <td colspan="4">
+                    <h3>Pods</h3>
```
Comment (dperpeet) [inline pkg/kubernetes/views/details-page.html]: translatable

File: pkg/kubernetes/views/images-page.html [hunk 20]
```diff
@@ -1,8 +1,10 @@
...
-    <table class="listing images-listing">
-        <thead ng-class="{selected: selected()}" ng-if="!quiet">
-            <tr class="listing-title">
-                <td colspan="4" translatable="yes">Image Repositories</td>
+<div ng-if="state">
+    <table listing-table class="listing images-listing">
+        <thead ng-if="!quiet">
+            <tr>
+                <td colspan="4">
+                    <h3>Image Repositories</h3>
```
Comment (dperpeet) [inline pkg/kubernetes/views/images-page.html]: translatable

File: pkg/kubernetes/views/images-page.html [hunk 21]
```diff
@@ -10,24 +12,23 @@
...
+            <tr ng-init="latest = item.latest()" ng-click="select(item.repo)"
+                ng-if="!quiet && is_visible(item)" data-id="{{ id }}" class="listing-item">
                 <th>{{item.name}}</th>
                 <td>{{item.tags() | imagesTags }}</td>
                 <td title="{{ latest }}">{{latest | timeAgo }}</td>
             </tr>
-            <tr class="listing-panel" ng-repeat-end ng-if="selected(item.repo)" data-id="{{ id }}">
-                <td cockpit-listing-panel kind="ImageRepository" colspan="3">
+            <tr class="listing-panel" ng-if="selected(item.repo)" data-id="{{ id }}">
+                <td listing-panel kind="ImageRepository" colspan="3">
```
Comment (dperpeet) [inline pkg/kubernetes/views/images-page.html]: close tag in same line?

File: pkg/kubernetes/views/replicationcontroller-panel.html [hunk 22]
```diff
@@ -1,64 +1,63 @@
-<div class="panel panel-default panel-tabbed"
-    ng-class="{ pinned: pinned }">
-    <div class="panel-heading">
-        <div class="listing-icons">
-            <i class="fa pficon-replicator"></i>
+<div class="listing-head" ng-click="select(id, $event)">
+    <div class="listing-actions">
+        <button title="Delete" class="delete-entity btn btn-danger btn-delete pficon pficon-delete"
```
Comment (dperpeet) [inline pkg/kubernetes/views/replicationcontroller-panel.html]: are titles automatically caught, or do we need translatable?

File: pkg/kubernetes/views/replicationcontroller-panel.html [hunk 23]
```diff
@@ -1,64 +1,63 @@
...
+            data-toggle="modal" data-target="#delete-entity-dialog" data-key="{{item.key}}"></button>
+        <button title="Adjust" class="adjust-entity btn btn-default pficon pficon-edit"
```
Comment (dperpeet) [inline pkg/kubernetes/views/replicationcontroller-panel.html]: are titles automatically caught, or do we need translatable?

File: pkg/kubernetes/views/replicationcontroller-panel.html [hunk 24]
```diff
@@ -1,64 +1,63 @@
...
+            data-toggle="modal" data-target="#adjust-entity-dialog" data-key="{{item.key}}"></button>
+    </div>
+    <h3><i class="fa pficon-replicator"></i>{{ item.metadata.name }}</h3>
+    <div class="listing-status">
+        <span ng-if="item.spec.replicas === item.status.replicas">{{ item.spec.replicas }}</span>
+        <span ng-if="item.spec.replicas !== item.status.replicas">{{ item.status.replicas }}
+            of {{ item.spec.replicas }}</span>
+    </div>
+    <ul class="nav nav-tabs nav-tabs-pf">
+        <li ng-class="{active: tab('main')}"><a ng-click="tab('main', $event)">Replication Controller</a></li>
```
Comment (dperpeet) [inline pkg/kubernetes/views/replicationcontroller-panel.html]: translatable

File: pkg/kubernetes/views/replicationcontroller-panel.html [hunk 25]
```diff
@@ -1,64 +1,63 @@
...
+        <li ng-class="{active: tab('pods')}"><a ng-click="tab('pods', $event)">Pods</a></li>
```
Comment (dperpeet) [inline pkg/kubernetes/views/replicationcontroller-panel.html]: translatable

File: pkg/kubernetes/views/replicationcontroller-panel.html [hunk 26]
```diff
@@ -1,64 +1,63 @@
...
+        <li ng-class="{active: tab('template')}"><a ng-click="tab('template', $event)">Template</a></li>
```
Comment (dperpeet) [inline pkg/kubernetes/views/replicationcontroller-panel.html]: translatable

File: pkg/kubernetes/views/route-panel.html [hunk 27]
```diff
@@ -1,48 +1,43 @@
...
-            <i class="pficon pficon-route"></i>
-        </div>
-        <div class="listing-actions">
-            <button title="Delete" class="btn btn-danger btn-delete pficon pficon-delete" disabled></button>
-            <button title="Adjust" class="btn btn-default pficon pficon-edit" disabled></button>
-        </div>
-        <h3 class="listing-title" ng-click="select(id)">{{ item.metadata.name }}</h3>
+<div class="listing-head" ng-click="select(id, $event)">
+    <div class="listing-actions">
+        <button title="Delete" class="btn btn-danger btn-delete pficon pficon-delete" disabled></button>
```
Comment (dperpeet) [inline pkg/kubernetes/views/route-panel.html]: are titles automatically caught, or do we need translatable?

File: pkg/kubernetes/views/route-panel.html [hunk 28]
```diff
@@ -1,48 +1,43 @@
...
-                        <dd ng-if="item.spec.to">{{item.spec.to.name}}</dd>
-                    </dl>
-                </div>
-                <div class="col-sm-6">
-                    <dl>
-                        <dt class="header full" translatable="yes">Labels</dt>
-                        <dd class="full" ng-repeat="(key, value) in item.metadata.labels">{{key}}={{value}}</dd>
-                        <dd class="full" ng-if="!item.metadata.labels" translatable="yes">none</dd>
+                <dt translatable="yes">TLS Termination</dt>
+                <dd ng-if="item.spec.tls.termination">yes</dd>
```
Comment (dperpeet) [inline pkg/kubernetes/views/route-panel.html]: translatable

File: pkg/kubernetes/views/route-panel.html [hunk 29]
```diff
@@ -1,48 +1,43 @@
...
+                <dd ng-if="!item.spec.tls.termination">no</dd>
```
Comment (dperpeet) [inline pkg/kubernetes/views/route-panel.html]: translatable

File: pkg/kubernetes/views/topology-page.html [hunk 30]
```diff
@@ -38,7 +38,7 @@ <h3 translatable="yes">Service</h3>
             <h3 translatable="yes">Node</h3>
             <p translatable="yes">Nodes are the machines that run your containers.</p>
         </div>
-        <div ng-if='selected && !(selected.kind === "Node")'>
+        <div class="topology-actions" ng-if='selected && !(selected.kind === "Node")'>
           <div ng-if='selected.kind === "Pod"'>
             <button id="delete-pod" title="Delete" class="btn btn-danger btn-delete pficon pficon-delete delete-entity " data-toggle="modal" data-target="#delete-pod-dialog" data-link="{{ selected.metadata.selfLink }}" ></button>
```
Comment (dperpeet) [inline pkg/kubernetes/views/topology-page.html]: translatable?

File: pkg/playground/patterns.html [hunk 31]
```diff
@@ -197,18 +210,364 @@ <h4 class="modal-title" translatable="yes">Example Dialog</h4>
...
-    "base1/bootstrap-select",
+    'jquery'
 ], function($) {
     "use strict";
 
-    $(document).ready(function() {
-        $('.selectpicker').selectpicker();
-        $("body").show();
+    $("body").on("click", ".listing-item, .listing-head", function(ev) {
+        if($(ev.target).parents().andSelf().filter(".btn, a, li, .timeline").length === 0) {
```
Comment (dperpeet) [inline pkg/playground/patterns.html]: see previous comment, [andSelf() is deprecated](https://api.jquery.com/andself/)
long chain of commands may merit a comment

File: test/check-kubernetes [hunk 32]
```diff
@@ -186,19 +189,19 @@ class TestKubernetes(KubernetesCase):
         b.click(".details-listing tbody[data-id='nodes/127.0.0.1'] th")
         b.wait_present(".details-listing tbody[data-id='nodes/127.0.0.1'] tr.listing-panel")
```
Comment (dperpeet) [inline test/check-kubernetes]: do we need a distinction here between present and visible? below, you changed the code to check for `is_visible`

**Comment by dperpeet** [comment]
didn't test this yet, just a cursory source code review

File: pkg/base1/cockpit.css (hunk 1 above)
Comment (stefwalter) [inline pkg/base1/cockpit.css]: Fixed.

File: pkg/base1/cockpit.css (hunk 2 above)
Comment (stefwalter) [inline pkg/base1/cockpit.css]: Fixed.

File: pkg/kubernetes/app.js (hunk 3 above)
Comment (stefwalter) [inline pkg/kubernetes/app.js]: Using addBack() with a comment.

File: pkg/kubernetes/views/container-panel.html (hunk 4 above)
Comment (stefwalter) [inline pkg/kubernetes/views/container-panel.html]: Fixed.

File: pkg/kubernetes/views/container-panel.html (hunk 5 above)
Comment (stefwalter) [inline pkg/kubernetes/views/container-panel.html]: Fixed.

File: pkg/kubernetes/views/container-panel.html (hunk 6 above)
Comment (stefwalter) [inline pkg/kubernetes/views/container-panel.html]: Fixed.

File: pkg/kubernetes/views/containers-page.html (hunk 7 above)
Comment (stefwalter) [inline pkg/kubernetes/views/containers-page.html]: Fixed.

File: pkg/kubernetes/views/containers-page.html (hunk 8 above)
Comment (stefwalter) [inline pkg/kubernetes/views/containers-page.html]: Changed to rubbish.

File: pkg/kubernetes/views/containers-page.html (hunk 9 above)
Comment (stefwalter) [inline pkg/kubernetes/views/containers-page.html]: Fixed.

File: pkg/kubernetes/views/containers-page.html (hunk 10 above)
Comment (stefwalter) [inline pkg/kubernetes/views/containers-page.html]: Fixed.

File: pkg/kubernetes/views/deploymentconfig-panel.html (hunk 11 above)
Comment (stefwalter) [inline pkg/kubernetes/views/deploymentconfig-panel.html]: Sadly attributes are not yet translatable.

File: pkg/kubernetes/views/deploymentconfig-panel.html (hunk 12 above)
Comment (stefwalter) [inline pkg/kubernetes/views/deploymentconfig-panel.html]: Fixed.

File: pkg/kubernetes/views/deploymentconfig-panel.html (hunk 13 above)
Comment (stefwalter) [inline pkg/kubernetes/views/deploymentconfig-panel.html]: Fixed.

File: pkg/kubernetes/views/details-page.html (hunk 14 above)
Comment (stefwalter) [inline pkg/kubernetes/views/details-page.html]: Fixed.

File: pkg/kubernetes/views/details-page.html (hunk 15 above)
Comment (stefwalter) [inline pkg/kubernetes/views/details-page.html]: Fixed.

File: pkg/kubernetes/views/details-page.html (hunk 16 above)
Comment (stefwalter) [inline pkg/kubernetes/views/details-page.html]: Fixed.

File: pkg/kubernetes/views/details-page.html (hunk 17 above)
Comment (stefwalter) [inline pkg/kubernetes/views/details-page.html]: Fixed.

File: pkg/kubernetes/views/details-page.html (hunk 18 above)
Comment (stefwalter) [inline pkg/kubernetes/views/details-page.html]: Fixed.

File: pkg/kubernetes/views/details-page.html (hunk 19 above)
Comment (stefwalter) [inline pkg/kubernetes/views/details-page.html]: Fixed

File: pkg/kubernetes/views/images-page.html (hunk 20 above)
Comment (stefwalter) [inline pkg/kubernetes/views/images-page.html]: Fixed.

File: pkg/kubernetes/views/images-page.html (hunk 21 above)
Comment (stefwalter) [inline pkg/kubernetes/views/images-page.html]: Fixed.

File: pkg/kubernetes/views/replicationcontroller-panel.html (hunk 22 above)
Comment (stefwalter) [inline pkg/kubernetes/views/replicationcontroller-panel.html]: Attributes cannot yet be translated.

File: pkg/kubernetes/views/replicationcontroller-panel.html (hunk 23 above)
Comment (stefwalter) [inline pkg/kubernetes/views/replicationcontroller-panel.html]: Titles cannot yet be translated.

File: pkg/kubernetes/views/replicationcontroller-panel.html (hunk 24 above)
Comment (stefwalter) [inline pkg/kubernetes/views/replicationcontroller-panel.html]: Fixed.

File: pkg/kubernetes/views/replicationcontroller-panel.html (hunk 25 above)
Comment (stefwalter) [inline pkg/kubernetes/views/replicationcontroller-panel.html]: Fixed.

File: pkg/kubernetes/views/replicationcontroller-panel.html (hunk 26 above)
Comment (stefwalter) [inline pkg/kubernetes/views/replicationcontroller-panel.html]: Fixed.

File: pkg/kubernetes/views/route-panel.html (hunk 27 above)
Comment (stefwalter) [inline pkg/kubernetes/views/route-panel.html]: Titles are not yet translatable.

File: pkg/kubernetes/views/route-panel.html (hunk 28 above)
Comment (stefwalter) [inline pkg/kubernetes/views/route-panel.html]: Fixed.

File: pkg/kubernetes/views/route-panel.html (hunk 29 above)
Comment (stefwalter) [inline pkg/kubernetes/views/route-panel.html]: Fixed.

File: pkg/kubernetes/views/topology-page.html (hunk 30 above)
Comment (stefwalter) [inline pkg/kubernetes/views/topology-page.html]: nope

File: pkg/playground/patterns.html (hunk 31 above)
Comment (stefwalter) [inline pkg/playground/patterns.html]: Using addBack() and added comment.

File: test/check-kubernetes (hunk 32 above)
Comment (stefwalter) [inline test/check-kubernetes]: In this case, .listing-panel is only present when visible.

**Comment by dperpeet** [comment]
When hovering over the listing pattern where header is the row and there are tabs, the tabs look a bit strange. Should we change the hover effect to be consistent with the other hover, where everything is highlighted? Or change the spacing? Right now the way the line between the two colors bisects the spacing between the lines looks strange.

listing_pattern_hover

**Comment by stefwalter** [comment]
> When hovering over the listing pattern where header is the row and there are tabs, the tabs look a bit strange. Should we change the hover effect to be consistent with the other hover, where everything is highlighted? Or change the spacing? Right now the way the line between the two colors bisects the spacing between the lines looks strange.

I guess we need to fix this in javascript.

**Comment by stefwalter** [comment]
> When hovering over the listing pattern where header is the row and there are tabs, the tabs look a bit strange. Should we change the hover effect to be consistent with the other hover, where everything is highlighted? Or change the spacing? Right now the way the line between the two colors bisects the spacing between the lines looks strange.

Fixed.

**Comment by dperpeet** [comment]
> Fixed.

looks a lot better now!

**Comment by dperpeet** [comment]
[playground] There's something fishy going on with the buttons in 'Expandable Listing: Panel header is row' when using Firefox 38.3.0: if you expand the second line, the rendered buttons in the line above seem to change their height. The calculated properties never change though. The issue is minor and I think it can be ignored for now.

**Comment by stefwalter** [comment]
Added dashed line as shown in the designs:

Dashed

**Comment by petervo** [comment]
On the kubernetes display page, the log and shell tabs are showing blank for me.

**Comment by andreasn** [comment]
I tested a bit on the Patterns page and looked good to me.

**Comment by stefwalter** [comment]
> On the kubernetes containers page, the log and shell tabs are showing blank for me.

This was a mistaken use of ng-if instead of ng-show. Fixed.

**Comment by petervo** [comment]
@stefwalter, thanks working now.

@dperpeet let me know if you are going to merge or if you want me to

**Pull request closed without merging**